core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -56,8 +56,12 @@ private[spark] abstract class YarnSchedulerBackend(
    * This includes executors already pending or running.
    */
   override def doRequestTotalExecutors(requestedTotal: Int): Boolean = {
+
+    val nodeBlacklist: Set[String] = scheduler.sc.blacklistTracker
+      .fold(Set.empty[String])(_.nodeBlacklist())
```

Comment:
I know this is normal scala code, but in spark we've decided to adopt a convention of *not* using `option.fold` -- instead we use `option.map(...).getOrElse(...)`, or maybe pattern matching depending on the case.  its just a little too confusing for those new to scala.
(you use this in a few other places too, I think)